modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -3343,11 +3343,12 @@ private void assignPartitionSizes(GridDhtPartitionTopology top) {
      * Collects and determines new owners of partitions for all nodes for given {@code top}.
      *
      * @param top Topology to assign.
+     * @param resetOwners True if need to reset partition state considering of counter, false otherwise.
      * @return Partitions supply info list.
      */
-    private List<SupplyPartitionInfo> assignPartitionStates(GridDhtPartitionTopology top) {
+    private List<SupplyPartitionInfo> assignPartitionStates(GridDhtPartitionTopology top, boolean resetOwners) {
         Map<Integer, CounterWithNodes> maxCntrs = new HashMap<>();
-        Map<Integer, T2<UUID, Long>> minCntrs = new HashMap<>();
+        Map<Integer, TreeSet<Long>> nonMaxCntrs = new HashMap<>();
```

Comment:
Now this map contains not only "nonMax" counters, but literally all various counters for each partition. We should come up with a better name.

Reply:
Renamed to varCntrs and replaced javadoc.